### PR TITLE
CPF-387 - Add unit tests for createUser/updateUser

### PR DESCRIPTION
**Pull request by greg-adams**
## Description
- Expands unit tests for `createUser` / `updateUser` mutations

### Automated and Unit Tests
- [x] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [x] Provided ticket and description
- [ ] Provided screenshots/demo
- [ ] Provided testing information
- [ ] Provided adequate test coverage for all new code
- [x] Added PR reviewers

### CODE REVIEW

File: api/setup-jest.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import initializeDb from './src/lib/db'
 
 beforeAll(async () => {
+  // avoid conflict in dev container
```
Comment (as1729) [inline api/setup-jest.ts:4]: I'm not sure this is needed based on the redwood docs here: https://docs.redwoodjs.com/docs/2.x/testing/#the-test-database

can you describe the issue you were noticing a bit more? Maybe we can find an alternate approach that solves it?

File: api/src/services/users/users.test.ts [hunk 2]
```diff
@@ -174,60 +184,179 @@ describe('user queries', () => {
...
+      const result = await updateUser({
+        id: original.id,
+        input: {
+          email,
+          name,
+          role,
+          agencyId: scenario.agency.one.id,
+          isActive,
+        },
+      })
```
Comment (as1729) [inline api/src/services/users/users.test.ts:235]: Removes use of `original` constant as it should be the same as `scenario.user.one`
```suggestion
      const result = await updateUser({
        id: scenario.user.one.id,
        input: {
          email,
          name,
          role,
          agencyId: scenario.agency.one.id,
          isActive,
        },
      })
```

File: api/src/services/users/users.test.ts [hunk 3]
```diff
@@ -174,60 +184,179 @@ describe('user queries', () => {
...
+      const result = await createUser({
+        input: {
+          email: 'uniqueemail2@test.com',
+          name: 'String',
+          agencyId: scenario.agency.one.id,
+          role: 'USDR_ADMIN',
+        },
+      })
+
+      expect(result.passageId).toEqual('new-id-1')
```
Comment (as1729) [inline api/src/services/users/users.test.ts:287]: Recommend checking whether the passage-user functions were called
```suggestion
      expect(result.passageId).toEqual('new-id-1')
      expect(mockPassageUser.create).to.haveBeenCalled()
      expect(mockPassageUser.activate).to.haveBeenCalled()
```

File: api/src/services/users/users.test.ts [hunk 4]
```diff
@@ -174,60 +184,179 @@ describe('user queries', () => {
...
+
+    scenario(
+      'updates a user - deactivate',
+      async (scenario: StandardScenario) => {
+        mockCurrentUser({
+          id: scenario.user.one.id,
+          email: scenario.user.one.email,
+          roles: ['USDR_ADMIN'],
+        })
+        const original = (await user({ id: scenario.user.one.id })) as User
```
Comment (as1729) [inline api/src/services/users/users.test.ts:298]:
```suggestion
        const original = scenario.user.one
```

File: api/src/services/users/users.test.ts [hunk 5]
```diff
@@ -174,60 +184,179 @@ describe('user queries', () => {
...
+        const passageUser = { id: 'new-id-1' }
+        mockPassageUser.create.mockReturnValue(passageUser)
+        mockPassageUser.activate.mockReturnValue(passageUser)
+
+        mockCurrentUser({
+          id: scenario.user.one.id,
+          email: scenario.user.one.email,
+          roles: ['USDR_ADMIN'],
+        })
+        const original = (await user({ id: scenario.user.inactive.id })) as User
```
Comment (as1729) [inline api/src/services/users/users.test.ts:328]:
```suggestion
        const original = scenario.user.inactive
```

File: api/src/services/users/users.test.ts [hunk 6]
```diff
@@ -174,60 +184,179 @@ describe('user queries', () => {
...
+            agencyId: original.agencyId,
+            isActive: true,
+          },
+        })
+
+        expect(result.isActive).toEqual(true)
+        expect(result.passageId).toEqual('new-id-1')
+        expect(mockPassageUser.create).toHaveBeenCalledWith({
+          email: original.email,
+        })
```
Comment (as1729) [inline api/src/services/users/users.test.ts:344]:
```suggestion
        })
        expect(mockPassageUser.activate).toHaveBeenCalled()
```

File: api/src/services/users/users.test.ts [hunk 7]
```diff
@@ -174,60 +184,179 @@ describe('user queries', () => {
...
+
+    scenario('deletes a user', async (scenario: StandardScenario) => {
+      mockCurrentUser({
+        id: scenario.user.one.id,
+        email: scenario.user.one.email,
+        roles: ['USDR_ADMIN'],
+      })
+      const original = (await deleteUser({
+        id: scenario.user.one.id,
+      })) as User
```
Comment (as1729) [inline api/src/services/users/users.test.ts:356]:
```suggestion
      const original = scenario.user.one
```

**Review by as1729 — COMMENTED** [review]
@greg-adams overall the PR looks good! I added a few suggestions but the only blocking point to me is the changes to `api/setup-jest.ts`. If you choose to accept the other changes I would recommend that we leave the changes in `api/setup-jest.ts` for a separate PR.

File: api/src/services/users/users.test.ts (hunk 7 above)
Comment (greg-adams) [reply api/src/services/users/users.test.ts:358]: leaving call to delete
```suggestion
      await deleteUser({
        id: scenario.user.one.id,
      }))
          expect(mockPassageUser.delete).toHaveBeenCalledWith(scenario.user.one.passageId)
```

File: api/src/services/users/users.test.ts [hunk 8]
```diff
@@ -174,60 +184,182 @@ describe('user queries', () => {
...
+    scenario('deletes a user', async (scenario: StandardScenario) => {
+      mockCurrentUser({
+        id: scenario.user.one.id,
+        email: scenario.user.one.email,
+        roles: ['USDR_ADMIN'],
+      })
+      await deleteUser({
+        id: scenario.user.one.id,
+      }))
+          expect(mockPassageUser.delete).toHaveBeenCalledWith(scenario.user.one.passageId)
```
Comment (greg-adams) [inline api/src/services/users/users.test.ts:359]:
```suggestion
expect(mockPassageUser.delete).toHaveBeenCalledWith(scenario.user.one.passageId)
```

File: api/src/services/users/users.test.ts [hunk 9]
```diff
@@ -174,60 +184,182 @@ describe('user queries', () => {
...
+        id: scenario.user.one.id,
+        email: scenario.user.one.email,
+        roles: ['USDR_ADMIN'],
+      })
+      await deleteUser({
+        id: scenario.user.one.id,
+      }))
+expect(mockPassageUser.delete).toHaveBeenCalledWith(scenario.user.one.passageId)
+
+      expect(mockPassageUser.delete).toHaveBeenCalledWith(original.passageId)
```
Comment (greg-adams) [inline api/src/services/users/users.test.ts:361]:
```suggestion
```

File: api/setup-jest.ts (hunk 1 above)
Comment (greg-adams) [reply api/setup-jest.ts]: I removed this change for now

**Review by as1729 — APPROVED** [review]

File: api/setup-jest.ts (hunk 1 above)
Comment (as1729) [reply api/setup-jest.ts]: @greg-adams can you open a separate PR with this change so we don't lose track of it? I'll have some time tomorrow to review and run the same tests locally to see what might be happening.

**Pull request merged**